docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,189 @@
+
+#### Given Members as LDAP Annotations on LDAP Groups
+If groups are first-class LDAP entries, then `LDAPCachedReferencedMemberLister` can query for all groups to be synced first. The generation of the LDAP filter for a specific group from the description of the LDAP group in the LDAP group entry will need to be specified by the user requesting a sync. If LDAP group members are returned by reference in an LDAP entry attribute, another LDAP filter will need to be created with the given information in order to find the LDAP first-class entry for the user, if it exists. The generation of the LDAP filter for a specific member from the description of the LDAP group member in the LDAP group member entry will need to be specified by the user requesting a sync. Caching can be implemented for LDAP responses representing first-class members or the `LDAPMemberLister` can be used for each group once the LDAP unique filter is created for it.
+
+### Collecting LDAP Members: Searching for one Group
+An `LDAPMemberLister` will be used inside of the sync job to do the first step in the sync: locate the LDAP entries that correspond to the group being synced, given the filter for that group:
+
+```go
+// LDAPMemberLister lists the LDAP entries that correspond to the members of the given Group
+type LDAPMemberLister interface {
+  ListMembers(groupFilter string) (members *[]ldap.Entry, err error)
+}
+```
+#### Given Members as First-Class LDAP Entries
+If members are first-class LDAP entries, the `LDAPFirstClassMemberLister` can simply query the LDAP server for members of a group and return those entries.
+
+#### Given Members as LDAP Annotations on LDAP Groups
+If LDAP group members are returned by reference in an LDAP entry attribute, another LDAP filter will need to be created with the given information in order to find the LDAP first-class entry for the user, if it exists. The generation of the LDAP query from the description of the LDAP group member will need to be specified by the user requesting a sync.
+
+### Mapping LDAP Members to OpenShift `Identities`
+This will be done in a similar way to the mapping for [LDAP password processing](https://github.com/openshift/origin/blob/master/pkg/auth/authenticator/password/ldappassword/ldap.go#L227-243) once the requisite data is collected.
+
+### Mapping OpenShift `Identities` Representing LDAP Members to OpenShift `Users`
+A new `UserIdentityMapper` will need to be created: the `ForcedDeterministicUserIdentityMapper`, which will assume that the unique identifier retrieved from the LDAP response will be directly equal to the unique identifier (`uid`) of the OpenShift `User` that matches this `Identity`. This is required as otherwise there is no deterministic mapping from the `Identity` created and any OpenShift `User`. This will need to be boldly documented.
```

Comment:
`ForcedDeterministicUserIdentityMapper` deserves its own spot in this doc describing how it will work and handle the edges.  I suspect we'll just want `DeterminsticUserIdentityMapper` with some reasonable collision options.